### PR TITLE
Exclude message types (mute join / leave messages in channel)

### PR DESCRIPTION
**Pull request by coltoneshaw**
#### Summary
This PR adds the ability to exclude specific message types when viewing channels. It adds `exclude_post_types` parameters to `user/{user_id}/channels/{chanel_id}/posts` and `/unreads`. 

This extends the functionality through `excludePostTypes` which accepts any `posts.type` into a `[]string`.

This PR is still a WIP to use the tests.

#### Release Note

```release-note
Adds the ability to exclude post types from channels.
```

#### Components for debate

- Right now this does not have a WS event to refresh the posts in the channel. I was concerned this could cause a flood of events for a non-urgent change that could be reflected on the next reload of MM. However, this could create a weird experience for the users and i can add it. 
- This sends a post into the channel saying it's been updated. I think the post could be modified or removed as desired.
Screenshot 2024-04-24 at 6 42 45 PM

#### UI Changes

https://github.com/mattermost/mattermost/assets/46071821/12216ea1-9328-41d4-8255-de87b84eff6e

### CODE REVIEW

File: server/channels/api4/post.go [hunk 1]
```diff
@@ -288,23 +298,23 @@ func getPostsForChannel(c *Context, w http.ResponseWriter, r *http.Request) {
...
+		list, err = c.App.GetPostsAfterPost(model.GetPostsOptions{ChannelId: channelId, PostId: afterPost, Page: page, PerPage: perPage, SkipFetchThreads: skipFetchThreads, CollapsedThreads: collapsedThreads, UserId: c.AppContext.Session().UserId, IncludeDeleted: includeDeleted, ExcludePostTypes: excludePostTypesList})
 	} else if beforePost != "" {
 		etag = c.App.GetPostsEtag(channelId, collapsedThreads)
 
 		if c.HandleEtag(etag, "Get Posts Before", w, r) {
 			return
 		}
 
-		list, err = c.App.GetPostsBeforePost(model.GetPostsOptions{ChannelId: channelId, PostId: beforePost, Page: page, PerPage: perPage, SkipFetchThreads: skipFetchThreads, CollapsedThreads: collapsedThreads, CollapsedThreadsExtended: collapsedThreadsExtended, UserId: c.AppContext.Session().UserId, IncludeDeleted: includeDeleted})
+		list, err = c.App.GetPostsBeforePost(model.GetPostsOptions{ChannelId: channelId, PostId: beforePost, Page: page, PerPage: perPage, SkipFetchThreads: skipFetchThreads, CollapsedThreads: collapsedThreads, CollapsedThreadsExtended: collapsedThreadsExtended, UserId: c.AppContext.Session().UserId, IncludeDeleted: includeDeleted, ExcludePostTypes: excludePostTypesList})
```
Comment (coltoneshaw) [inline server/channels/api4/post.go:309]: There is the change, i believe, this is returned as `nil`.

**Comment by agnivade** [comment]
@coltoneshaw - Just checking, is this ready to be reviewed?

**Comment by coltoneshaw** [comment]
> @coltoneshaw - Just checking, is this ready to be reviewed?

Yes, it is! However, I was requested to change the channel drop down before this can be considered. https://github.com/mattermost/mattermost/issues/26734

I, unfortunately, do not have time to address that right now. So if we need to close this PR we can.

**Comment by agarciamontoro** [comment]
We should probably close this one and re-open when needed.

**Pull request closed without merging**